pkg/controller/template/template_controller.go

```diff
@@ -103,6 +112,59 @@ func New(
 	return ctrl
 }
 
+func (ctrl *Controller) filterSecret(secret *v1.Secret) {
+	if secret.GetName() == "pull-secret" {
+		cfg, err := ctrl.ccLister.Get(common.ControllerConfigName)
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("couldn't get ControllerConfig on secret callback %#v", err))
+			return
+		}
+		glog.V(4).Infof("Re-syncing ControllerConfig %s due to secret change", cfg.Name)
+		ctrl.enqueueControllerConfig(cfg)
+	}
+}
+
+func (ctrl *Controller) addSecret(obj interface{}) {
+	secret := obj.(*v1.Secret)
+	if secret.DeletionTimestamp != nil {
+		ctrl.deleteSecret(secret)
+		return
+	}
+	ctrl.filterSecret(secret)
+}
+
+func (ctrl *Controller) updateSecret(old, new interface{}) {
```

Comment:
updateSecret only seems to call filterSecret and only uses 1 of its args (just new not old)?

Reply:
yup, we don't really care about `old` and we filter the correct one in `filterSecret` anyway so `old` can be discarded. We can't really use `filterSecret` in the event handler for the informer cause it's not type casted to secret automatically (which is what this method does eventually).

Reply:
gotcha on the filterSecret vs updateSecret part :+1:  but not understanding why old is an arg for updateSecret if we don't need it or have the logic? shouldn't unused args throw some kind of golang linting error? won't it do that in my vscode?

Reply:
that's an interface for the event handler for update. The interface for the informer callbacks are as follows:
```
type ResourceEventHandlerFuncs struct {
	AddFunc    func(obj interface{})
	UpdateFunc func(oldObj, newObj interface{})
	DeleteFunc func(obj interface{})
}
```
You can see `UpdateFunc` has old and new (as an update action should have). `updateSecret` is using the `UpdateFunc` signature so we must have old and new.

Anyway, for linting golang doesn't care and vscode either (at least not mine). Golang has the don't care underscore for not used args/variable but that generally sucks, i.e. the updateSecret func becomes

```
func updateSecret(_, new interface{}) {
```

Reply:
thank you for explaining, i didn't quite get it the first two times around :smiley: